gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PluginController.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.controllers;
+
+import com.netflix.spinnaker.gate.services.TaskService;
+import com.netflix.spinnaker.gate.services.internal.Front50Service;
+import com.netflix.spinnaker.security.AuthenticatedRequest;
+import io.swagger.annotations.ApiOperation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/pluginInfo")
+public class PluginController {
+
+  // TODO: Make it configurable.
+  private static final String DEFAULT_APPLICATION = "spinnaker";
```

Comment:
Any reason not to make this configurable right now?  A config with the default value of `spinnaker` would allow us to remove this TODO and give us a bit more flexibility.

Reply:
I'd say we should make this configurable now. Also, I think we should make the default something a little more unique, like, `spinnakerplugins`. After talking with @cfieber, I think it would be easier to add a security model if we didn't run the potential to clash with existing applications. For example, a basic initial security strategy might be that we just require people to have `WRITE` to a specific application to publish plugins. For the OSS community, I imagine a lot of people have single-stack Spinnaker installs under the `spinnaker` application name and that would open them up to a wider set of people having access to Spinnaker itself - not good. For us at Netflix, we have a similar case where our proxy instances live under `spinnaker` and we don't want to grant other people access to that app either.

Reply:
Made the app name configurable and applied security aspects around all write operations.